hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationRegistry.java

```diff
@@ -319,28 +333,50 @@ private void scanHandleOperationTimeout() {
                     return;
                 }
 
-                try {
-                    if (invocation.checkInvocationTimeout()) {
-                        invocationTimeouts++;
-                    }
-                } catch (Throwable t) {
-                    inspectOutputMemoryError(t);
-                    logger.severe("Failed to handle operation timeout of invocation:" + invocation, t);
+                logSlowInvocation(now, invocation);
+
+                if (checkInvocationTimeout(invocation)) {
+                    invocationTimeouts++;
                 }
 
-                try {
-                    if (invocation.checkBackupTimeout(backupTimeoutMillis)) {
-                        backupTimeouts++;
-                    }
-                } catch (Throwable t) {
-                    inspectOutputMemoryError(t);
-                    logger.severe("Failed to handle backup timeout of invocation:" + invocation, t);
+                if (checkBackupTimeout(invocation)) {
+                    backupTimeouts++;
                 }
             }
 
             log(backupTimeouts, invocationTimeouts);
         }
 
+        private boolean checkBackupTimeout(Invocation invocation) {
+            try {
+                return invocation.checkBackupTimeout(backupTimeoutMillis);
+            } catch (Throwable t) {
+                inspectOutputMemoryError(t);
+                logger.severe("Failed to handle backup timeout of invocation:" + invocation, t);
+                return false;
+            }
+        }
+
+        private boolean checkInvocationTimeout(Invocation invocation) {
+            try {
+                return invocation.checkInvocationTimeout();
+            } catch (Throwable t) {
+                inspectOutputMemoryError(t);
+                logger.severe("Failed to handle operation timeout of invocation:" + invocation, t);
+                return false;
+            }
+        }
+
+        private void logSlowInvocation(long now, Invocation invocation) {
+            if (slowInvocationThresholdMs > 0) {
+                long durationMs = now - invocation.op.getInvocationTime();
```

Comment:
However.. There is one thing that might need some reconsideration. And that the invocation time is set using cluster lock, but timeout checks are done using Clock... and Cluster clock can skew.